### PR TITLE
Pipeline changes per Mike's suggestions

### PR DESCRIPTION
**Pull request by dmarginian**

### CODE REVIEW

File: core/impl/main/java/dwr-version.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 major=3
 minor=0
-revision=3-SNAPSHOT
-title=dev
+revision=3
+title=SNAPSHOT
+# build.number property is added by CI builds
 build.number=
```
Comment (mikewse) [inline core/impl/main/java/dwr-version.properties:6]: Remove the "build.number=" line and just leave the comment above.

File: build.xml [hunk 2]
```diff
@@ -7,8 +7,11 @@
 	<!-- File that contains local settings, you may need to edit this -->
 	<property file="build.properties" />
 
-	<!-- environment used during CI -->
-	<property environment="env"/>
+	<!-- Get build number from Github CI -->
+	<property environment="envgithub"/>
+	<condition property="build.number">
+		<isset property="envgithub.GITHUB_RUN_ID" />
+	</condition>
```
Comment (mikewse) [inline build.xml:14]: I think you forgot to set the build.number value?

Comment (dmarginian) [reply build.xml]: I re-worked this a bit.

File: build.xml [hunk 3]
```diff
@@ -33,6 +27,21 @@
...
+		<if>
+			<isset property="envgithub.GITHUB_RUN_ID" />
+			<then>
+				<propertyfile file="${target.root}/classes/main/dwr-version.properties">
+					<entry key="build.number" value="${envgithub.GITHUB_RUN_ID}" />
+				</propertyfile>
+			</then>
+		</if>
+	</target>
+
```
Comment (mikewse) [inline build.xml:44]: Ok so you are now back at roughly the same solution as I initially commented on. This is not a big deal so do like you want, but the reason I preferred that you do:
  build.number = envgithub.GITHUB_RUN_ID
in the property section and then use build.number, is to isolate as much as possible of build.xml from external system dependencies. In theory we could get a build number from any place and then we would just edit the property assignment. Using envgithub.GITHUB_RUN_ID or any other envgithub.<var> inside target code is less separated IMO even if those targets are pushed to top of the file.

Comment (mikewse) [inline build.xml:44]: Ok so you are now back at roughly the same solution as I initially commented on. This is not a big deal so do like you want, but the reason I preferred that you do:
build.number = envgithub.GITHUB_RUN_ID
in the property section and then use build.number, is to isolate as much as possible of build.xml from external system dependencies. In theory we could get a build number from any place and then we would just edit the property assignment. Using envgithub.GITHUB_RUN_ID or any other envgithub.VAR inside target code is less separated IMO even if those targets are pushed to top of the file.

Comment (dmarginian) [reply build.xml]: Ok Mike, we should be good now.  Take another look.

**Pull request merged**